packages/fuels-core/src/types/bits.rs

```diff
@@ -90,16 +144,22 @@ impl Tokenizable for EvmAddress {
     where
         Self: Sized,
     {
-        if let Token::Struct(tokens) = &t {
-            let first_token = tokens.iter().next();
+        if let Token::Struct(tokens) = t {
+            let first_token = tokens.into_iter().next();
             if let Some(Token::B256(data)) = first_token {
-                return Ok(EvmAddress::from(Bits256(*data)));
+                Ok(EvmAddress::from(Bits256(data)))
+            } else {
+                Err(Error::InstantiationError(format!(
+                    "Expected `b256`, got {:?}",
```

Comment:
Technically we could be more strict in checking the tokens against our expectations. We should be given a struct with only one token inside of it. Here we're willing to accept multiple tokens and just consume the first one.

If we ever got multiple tokens here it would be a good sign that something is broken.

Also, when informing that the first token was not of the right type, we don't mention that there happened to be others. Information that might be useful since these would probably be the tokens of some other struct. 

Maybe there is a way to get the full error message and take ownership of the first element?

Reply:
Agreed that we should check that length is exactly one